app/models/user.rb

```diff
@@ -1,6 +1,64 @@
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
-  devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :validatable
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable, :registerable,
+  devise :invitable, :database_authenticatable,
+         :recoverable, :rememberable, :validatable, invite_for: 2.weeks
+
+  has_many :taught_classes, dependent: :destroy
+  has_many :assignments, through: :taught_classes
+  has_many :class_memberships, dependent: :destroy
+  has_many :student_answers, through: :completed_assignments
+  has_many :completed_assignments
+
+  belongs_to :school
+
+
+  enum role: [:teacher, :student, :admin]
+  after_initialize :set_default_role, :if => :new_record?
+
+  scope :students, -> { where role: "student" }
+  scope :teachers, -> { where role: "teacher" }
+  scope :admins, -> { where role: "admin" }
+
+  accepts_nested_attributes_for :school
+
+
+
+  validates :name, presence: true
+  # validates :role, presence: true
```

Comment:
If you add a presence-validator, it is good practice to also set `null: false` in the database. 

Here's some background info: https://thoughtbot.com/blog/validation-database-constraint-or-both